site/tutorial/07-ch05-filter-service.markdown

```diff
@@ -279,71 +279,53 @@ void main() {
 <p>Next, let’s look at how to use the
 Http service to fetch
 data from the server. Look at the changes we made to the
-<code>_loadData</code> method. Here is the new code:</p>
+<code>_loadData()</code> method. Here is the new code:</p>
 
 <script type="template/code">
-void _loadData() {
+Future _loadData() async {
   recipesLoaded = false;
   ...
-  _http.get('recipes.json')
-    .then((HttpResponse response) {
+try {
+      var response = await _http.get('recipes.json');
+      print(response);
       recipes = response.data.map((d) => new Recipe.fromJson(d)).toList();
       recipesLoaded = true;
-    })
-    .catchError((e) {
-      recipesLoaded = false;
-      message = ERROR_MESSAGE;
-    });
+    } on Error catch (e) {
+        print(e);
+        recipesLoaded = false;
+        message = ERROR_MESSAGE;
+    };
   ...
 }
 </script>
 
-<p>Let’s look more closely at the call to the <code>Http</code> service:</p>
+<p>Using HTTP to fetch data from a server introduces a problem that's
+familiar in web-app development: if the HTTP connection is slow, the
+app could hang until the data is returned.</p>
 
-<script type="template/code">
-_http.get(URL)
-  .then((value) {
-    // use value
-  })
-  .catchError((e) {
-    // handle error
-  });
-</script>
-
-<p>The <code>Http</code> <code>get</code> method returns a
+<p>To avoid this problem, the <code>_loadData()</code> method declares
+itself to be asynchronous by using the <code>async</code> modifier
+and immediately returning a 
 <a href="http://api.dartlang.org/docs/releases/latest/dart_async/Future.html">
   Dart Future</a>. A Future is the promise of a value sometime in the future.
 It is at the core of
 <a href="https://www.dartlang.org/docs/dart-up-and-running/contents/ch03.html#ch03-asynchronous-programming">
-  asynchronous programming in Dart</a>. In its simplest form, the
-<code>then()</code> method  takes a single function argument. This
-function is invoked when the Future completes with a value (i.e.,
-successfully), and is used to process the value returned from the
-Future. The <code>catchError()</code> method also takes a function
-argument. This function  will be invoked if an error is emitted by the
-Future.</p>
-
-<p>The important thing to understand from this example is that Futures are
-asynchronous. Your app code will proceed while the data is loading from
-the server side. If you are connecting to a very slow service, it is
-possible that the app will finish loading before the data has been
-returned. The view should be prepared for this. In our case, we need two
-pieces of data before the app is useful:</p>
-
+  asynchronous programming in Dart</a>. In its simplest form, an asynchronous
+method contains at least one <code>await</code> expression, which pauses execution
+of all subsequent expressions until the <code>await</code> expression completes.</p>
```

Comment:
Perhaps acknowledge that an async method doesn't even execute a single one of its statements before returning a Future?

Also, add a blank line between `<p>`s.